### PR TITLE
Disable turnstile globally, or if redis is selected for storage

### PR DESCRIPTION
**Pull request by maccabeelevine**

### CODE REVIEW

**Review by demiankatz — APPROVED** [review]
Makes sense to me, but I'll wait for this to come out of draft and get feedback from @EreMaijala before merging it.

**Comment by maccabeelevine** [comment]
Even if we do support redis when installed (which is now on my todo list), it's also a win here remove the constructor dependency as @demiankatz suggested, and also to make it easier to globally disable turnstile in one place if you have multiple policies, and to avoid the minor resource usage of constructing the turnstile cache if it's not actually going to be needed.

File: module/VuFind/src/VuFind/RateLimiter/RateLimiterManager.php
Comment (EreMaijala) [inline module/VuFind/src/VuFind/RateLimiter/RateLimiterManager.php]: This should be `@var ?Turnstile`

**Review by EreMaijala — CHANGES_REQUESTED** [review]
Looks good apart from the nitpicky comment issue.

File: module/VuFind/src/VuFind/RateLimiter/RateLimiterManager.php
Comment (demiankatz) [inline module/VuFind/src/VuFind/RateLimiter/RateLimiterManager.php]:
```suggestion
     * @var ?Turnstile
```

**Review by EreMaijala — APPROVED** [review]

**Pull request merged**
